# Author: Swetha Daddala, Tanvi Sandeep Bhagat, Ketaki Joshi, Yuanying Li

# Purpose: Sentiment analysis 

# Course: ALY 6030 


In [1]:
!pip install textblob

Importing TextBlob to dive into "Natural Language Processing (NLP)" for Sentiment Analysis.

Importing pandas for data manipulation and analysis.

Importing io package to handle file related input/output operations.

In [2]:
from textblob import TextBlob 
import pandas as pd
import io

Loading the csv file from local storage using read_csv

In [3]:
url = 'https://docs.google.com/spreadsheets/d/1M29kDEmXpDbxfBjbCUpdIyoKbQfSvMQQRgY5Dm07zWQ/edit?usp=sharing'
path = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vTTTjDc4gkwlHZ4Wit1fcBffneXFGu9yc96iR5MxR_bwwWC_PuOtqQw2RcYnMZdJh7N32XTxSHSy2FG/pub?gid=1383303828&single=true&output=csv'
orders_df = pd.read_csv(path)

Importing the necessary packages

Reading the data into the "orders_df" dataframe

Viewing the data from the "orders_df" dataframe

In [4]:
#orders_df = pd.read_csv(io.BytesIO(uploaded['orders.csv']))
# Dataset is now stored in a Pandas Dataframe
print(orders_df) #Viewing the data in the dataframe

    order_id  customer_id  employee_id  time_key  \
0        900        10001          104         1   
1        901        10002          104         1   
2        902        10002          105         2   
3        903        10009          105         5   
4        904        10005          106         8   
5        905        10004          107         4   
6        906        10014          108         1   
7        907        10008          146        30   
8        908        10003          117       100   
9        909        10020          177       110   
10       910        10016          116       121   
11       911        10015          179        10   
12       912        10011          179        45   
13       913        10005          109        41   
14       914        10008          110        90   
15       915        10010          101        98   
16       916        10012          100        33   
17       917        10013          106        78   
18       918

Converting the Upper Case letters and upper case wordss to lower case using "str.lower()" function


In [5]:
orders_df["comments"] = orders_df["comments"].str.lower() #Converting the Uppercase letters to lower case
print(orders_df)

    order_id  customer_id  employee_id  time_key  \
0        900        10001          104         1   
1        901        10002          104         1   
2        902        10002          105         2   
3        903        10009          105         5   
4        904        10005          106         8   
5        905        10004          107         4   
6        906        10014          108         1   
7        907        10008          146        30   
8        908        10003          117       100   
9        909        10020          177       110   
10       910        10016          116       121   
11       911        10015          179        10   
12       912        10011          179        45   
13       913        10005          109        41   
14       914        10008          110        90   
15       915        10010          101        98   
16       916        10012          100        33   
17       917        10013          106        78   
18       918

Applying Lambda function on comments column to notice positive sentiment among customers. For this, we search for comments like "good, great, like, best, enjoy."

In [6]:
searchfor = ['good', 'great', 'like', 'best', 'enjoy','must', 'love', 'clean']
orders_df["positive_sentiment"] = orders_df['comments'].apply(lambda x: 1 if any(i in x for i in searchfor) else 0) 
# print(orders_df)



Performing Basic Sentiment Analysis:

The sentiment property returns a namedtuple of the form Sentiment(polarity, subjectivity). 

The polarity score is a float within the range [-1.0, 1.0]. 

The subjectivity is a float within the range [0.0, 1.0] where 0.0 is very objective and 1.0 is very subjective.

In [7]:
orders_df[['polarity', 'subjectivity']] = orders_df['comments'].apply(lambda Text: pd.Series(TextBlob(Text).sentiment)) #Checking polarity and subjectivity using sentiment property
print(orders_df)

    order_id  customer_id  employee_id  time_key  \
0        900        10001          104         1   
1        901        10002          104         1   
2        902        10002          105         2   
3        903        10009          105         5   
4        904        10005          106         8   
5        905        10004          107         4   
6        906        10014          108         1   
7        907        10008          146        30   
8        908        10003          117       100   
9        909        10020          177       110   
10       910        10016          116       121   
11       911        10015          179        10   
12       912        10011          179        45   
13       913        10005          109        41   
14       914        10008          110        90   
15       915        10010          101        98   
16       916        10012          100        33   
17       917        10013          106        78   
18       918

CONCLUSION: 

Polarity & Subjectivity Interpretation for top3: 

order_id 900 -
0.44 defines the polarity of the sentence is slightly positive and subjectivity value 0.53 indicates it is fairly subjective.

order_id 901 - 
0.50 indicates neutral positivity and subjectivity value 0.675 indicates that the statement is fairly subjective.

order_id 903 - 
0.0 indicates that the statement shows neutral polarity and Very Objective.